mozilla_django_oidc_db/setup_config.py

```diff
@@ -0,0 +1,83 @@
+from django.conf import settings
+from django.contrib.sessions.middleware import SessionMiddleware
+from django.test import RequestFactory
+from django.utils.module_loading import import_string
+from django.utils.translation import gettext as _
+
+from django_setup_configuration.config_settings import ConfigSettings
+from django_setup_configuration.configuration import BaseConfigurationStep
+from django_setup_configuration.exceptions import ConfigurationRunFailed, SelfTestFailed
+
+from .forms import OIDCSetupConfigForm
+from .models import OpenIDConnectConfig
+from .utils import create_missing_groups
+
+
+class AdminOIDCConfigurationStep(BaseConfigurationStep):
+    """
+    Configure admin login via OpenID Connect
+    """
+
+    verbose_name = _("Configuration for admin login via OpenID Connect")
+
+    config_settings = ConfigSettings(
+        enable_setting="OIDC_DB_CONFIG_ENABLE",
+        display_name=_("Admin OIDC Configuration"),
+        namespace="OIDC_DB",
+        models=[OpenIDConnectConfig],
+        update_fields=True,
+        required_settings=["OIDC_DB_SETUP_CONFIG_ADMIN_AUTH"],
+    )
+
+    def is_configured(self) -> bool:
+        return OpenIDConnectConfig.get_solo().enabled
+
+    def configure(self):
+        config = OpenIDConnectConfig.get_solo()
+
+        all_settings = {
+            "sync_groups": config.sync_groups,
+            "oidc_use_nonce": config.oidc_use_nonce,
+            "enabled": True,
+            "claim_mapping": config.claim_mapping,  # JSONFormField widget cannot handle blank values with object schema
+            "sync_groups_glob_pattern": config.sync_groups_glob_pattern,
+            **settings.OIDC_DB_SETUP_CONFIG_ADMIN_AUTH,
+        }
```

Comment:
Not sure if  we should set the default values here or take them from the model like it currently does